media/js/src/exerciseSpace.tsx

```diff
@@ -25,39 +26,74 @@ export const ExerciseSpace: React.FC = () => {
     const [questionStatus, setQuestionStatus] = useState('');
     const [stepList, setStepList] = useState<[string, string][]>([]);
     const [hint, setHint] = useState<[string, string]>(['','']);
-    // user's proposed next step
-    const [nextStep, setNextStep] = useState('');
-    // user's proposed rule
-    const [nextRule, setNextRule] = useState('');
-    const [toolsResp, setToolsResp] = useState<Tools[]>([]);
+    const [nextStep, setNextStep] = useState('');  // user's proposed next step
+    const [nextRule, setNextRule] = useState('');  // user's proposed rule
     const [hintButtonCount, setHintButtonCount] = useState<number>(0);
 
     async function fetchStatement() {
         // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
         const json: Statement = await getStatement(Number(id));
         setStatement(json);
     }
-
     async function fetchSolutions() {
         // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
         const json: Array<Solution> = await getSolutions(Number(id));
         setSolutions(json);
     }
+    const getQuestionData = () => {
+        try {
+            const data = JSON.parse(
+                window.localStorage.getItem(
+                    'question-' + id)) as ExerciseData[];
+            const questStatus = data[0].status;
+            const stepList = data[0].stepList;
+            setQuestionStatus(questStatus);
+            setStepList(stepList);
+        } catch (error) {
+            setQuestionStatus(null);
+        }
+    };
+    async function fetchHints() {
+        const hintData: HintData = {
+            next_expr: '',
+            rule: '',
+            step_list: [''],
+            answer: ''
+        };
+        //Only if we don't already have hints
+        if(hintButtonCount === 0) {
```

Comment:
Check the eslint settings here - should be a space between the conditional and the (, e.g. `if (hintButtonCount === 0`